diskquota.c

```diff
@@ -640,15 +640,52 @@ disk_quota_launcher_main(Datum main_arg)
 	/* main loop: do this until the SIGTERM handler tells us to terminate. */
 	ereport(LOG, (errmsg("[diskquota launcher] start main loop")));
 	curDB = NULL;
-
 	while (!got_sigterm)
 	{
-		int  rc;
+		int rc;
+		CHECK_FOR_INTERRUPTS();
+
+		/*
+		 * modify wait time
+		 */
+		long secs;
+		int  microsecs;
+		TimestampDifference(GetCurrentTimestamp(),
+		                    TimestampTzPlusMilliseconds(loop_start_time, diskquota_naptime * 1000L), &secs, &microsecs);
+		nap.tv_sec  = secs;
+		nap.tv_usec = microsecs;
+
+		if (curDB == DiskquotaLauncherShmem->dbArrayTail)
+		{
+			/* Have sleep enough time, should start another loop */
+			if (nap.tv_sec == 0 && nap.tv_usec == 0)
+			{
+				loop_start_time = GetCurrentTimestamp();
+				/* set the curDB pointing to the head of the db list */
+				curDB = NULL;
+			}
+			/* do nothing, just to sleep untill the nap time is 0 */
+			else
+			{
+				continue;
```

Comment:
```suggestion
				pg_sleep(0.1);
				continue;
```

What about adding `pg_sleep(little_amount_of_time)` to this loop to reduce some CPU consumption?

Reply:
> What about adding `pg_sleep(little_amount_of_time)` to this loop to reduce some CPU consumption?

Hmm... this looks not helpful to resolve the issue.

Reply:
I mean if the naptime is 1s and `curDB == DiskquotaLauncherShmem->dbArrayTail` (the launcher have finished one loop), the launcher will keep looping here and do nothing, the load of the CPU that occupied by the launcher process will be 100% for 1s.

I don't insist on this and please correct me if I'm wrong.